File: longest-palindrome/longest-palindrome.cs
public class Solution {
    public int LongestPalindrome(string s) {
        int[] count = new int[128];
        
        foreach(char c in s)
        {
            count[c]++;
           
        }
        int ans =0;
        
        foreach(int v in count)
        {
             //Console.WriteLine(v/2*2);
            ans+= v/2*2;
            if(ans %2==0 && v%2==1)
            {
                ans++;
            }
        }
        return ans;
        
    }
}

File: assignments/src/Longest_Palindrome.java
public class Longest_Palindrome {
    // Time: O(n), Space: O(1)
    public int longestPalindrome(String s) {
        int charCount[] = new int[128];
        int ans = 0;

        for (char c : s.toCharArray())
            charCount[c]++;

        for (int count : charCount) {
            ans += count / 2 * 2;
            if (ans % 2 == 0 && count % 2 == 1)
                ans++;
        }
        return ans;
    }
}
